src/service-override/views.ts

```diff
@@ -304,17 +340,39 @@ function registerEditorPane (options: EditorPanelOption): { disposable: IDisposa
     }
   }
 
-  const disposable = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
+  const disposableStore = new DisposableStore()
+
+  disposableStore.add(Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
     EditorPaneDescriptor.create(
       CustomEditorPane,
       options.id,
       options.name
     ),
-    [new SyncDescriptor(CustomEditorInput)])
+    [new SyncDescriptor(CustomEditorInput)]))
 
   return {
-    disposable,
-    CustomEditorInput
+    disposable: disposableStore,
+    CustomEditorInput,
+    async registerEditor (globPattern: string, registerEditorOptions: RegisterEditorOptions = {}): Promise<IDisposable> {
```

Comment:
At the end, that's a shame to return a Promise here because it's harder to handle, you need to store the promise and await/handle error when you want to dispose it.

I don't see any reason to call that api before the services are initialized anyway, what about checking that the services are initialized (throw an error otherwise?) instead and use `StandaloneServices.get`